pkg/operator/controller/ingressclass/ingressclass.go

```diff
@@ -59,7 +73,11 @@ func (r *reconciler) ensureIngressClass(ingresscontrollerName string, ingresscla
 
 // desiredIngressClass returns a Boolean indicating whether an IngressClass
 // is desired, as well as the IngressClass if one is desired.
-func desiredIngressClass(ingresscontrollerName string, ingressclasses []networkingv1.IngressClass) (bool, *networkingv1.IngressClass) {
+func desiredIngressClass(haveIngressController bool, ingresscontrollerName string, ingressclasses []networkingv1.IngressClass) (bool, *networkingv1.IngressClass) {
+	if !haveIngressController {
```

Comment:
Doesn't have to happen in this PR, but should we consider adding some unit test coverage for `desiredIngressClass` and `ingressclassChanged`? Speaking of which, at some point we should change `ingressclassChanged` to `ingressClassChanged` as well :grin:

Reply:
Added unit tests and fixed capitalization.  Thanks!